src/apps/tetris.js

```diff
@@ -0,0 +1,152 @@
+import { useEffect, useRef } from "react";
+import Tetris from "react-tetris";
+import styled from "styled-components";
+import { matchTriggerQueries } from "../lib/utilityApis";
+
+const Wrapper = styled.section`
+  .game-block {
+    margin: 0;
+    padding: 0;
+    width: 1.5em;
+    height: 1.5em;
+    border: 1px solid #ddd;
+  }
+  .piece-i {
+    background-color: #ec858b;
+  }
+  .piece-j {
+    background-color: #f1b598;
+  }
+  .piece-l {
+    background-color: #f8efae;
+  }
+  .piece-o {
+    background-color: #b5a677;
+  }
+  .piece-s {
+    background-color: #816e56;
+  }
+  .piece-t {
+    background-color: #b77c72;
+  }
+  .piece-z {
+    background-color: #e3be58;
+  }
+  .piece-preview {
+    background-color: #eee;
+  }
+`;
+
+const StyledContainer = styled.div`
+  display: grid;
+  grid-template-columns: auto auto auto;
+  background-color: #080909;
+  padding: 10px;
+`;
+
+const StyledItem = styled.div`
+  background-color: #aaaaaa;
+  border: 1px solid rgba(0, 0, 0, 0.8);
+  padding: 10px;
+  font-size: 1rem;
+  text-align: center;
+`;
+const StyledHidden = styled.div`
+  position: absolute;
+  height: 100px;
+  width: 100px;
+  right: -1000px;
+  top: 50px;
+`;
+const StyledHiddenOuter = styled.div`
+  overflow: hidden;
+  position: relative;
+`;
+
+function Renderer() {
+  const useFocus = () => {
+    const mainRef = useRef(null);
+    const setFocus = (condition = true) => {
+      if (condition) {
+        return mainRef.current && mainRef.current.focus();
+      } else {
+        return mainRef.current && mainRef.current.blur();
+      }
+    };
+    return [mainRef, setFocus];
+  };
+  const [mainRef, setMainRef] = useFocus();
+
+  useEffect(() => {
+    setMainRef();
+    setMainRef(false);
+  }, []);
+  window.addEventListener("keydown", function (e) {
+    if (e.code === "Space" && e.target === document.body) {
+      e.preventDefault();
+    }
+    if (e.code === "ArrowUp" && e.target === document.body) {
+      e.preventDefault();
+    }
+    if (e.code === "ArrowDown" && e.target === document.body) {
+      e.preventDefault();
+    }
+  });
+  return (
+    <>
+      <StyledHiddenOuter>
+        <StyledHidden>
+          <input
+            type="text"
+            name="name"
+            STYLE=" background-color: #aaaaaa;border: transparent"
+            ref={mainRef}
+            tabindex="-1"
+          />
+        </StyledHidden>
+      </StyledHiddenOuter>
```

Comment:
What is the purpose of these lines here?

Reply:
The game controls are arrow keys and space bar, The other games in the app didn't require switching focus off the search box since they used WASD keys. To create similar effect in this app it was necessary to prevent default arrow key to cursor control behavior and scroll behavior. a hidden text box is introduced to divert focus from search box and blurred immediately. Also default behavior is prevented to avoid scrolling.

Reply:
Thanks for explaining, just add a few comments explaining things and then we can merge :smile:

Reply:
Added Comments